src/virtual-repeat.js

```diff
@@ -271,26 +271,39 @@ export class VirtualRepeat extends AbstractRepeater {
   _getMore(): void {
     if (this.isLastIndex || this._first === 0) {
       if (!this._calledGetMore) {
-        let getMoreFunc = this.view(0).firstChild.getAttribute('virtual-repeat-next');
-        if (!getMoreFunc) {
-          return;
-        }
-        let getMore = this.scope.overrideContext.bindingContext[getMoreFunc];
         let executeGetMore = () => {
           this._calledGetMore = true;
-          if (getMore instanceof Promise) {
-            return getMore.then(() => {
-              this._calledGetMore = false; //Reset for the next time
-            });
-          } else if (typeof getMore === 'function') {
-            let result = getMore.bind(this.scope.overrideContext.bindingContext)(this._first, this._bottomBufferHeight === 0, this._isAtTop);
-            if (!(result instanceof Promise)) {
-              this._calledGetMore = false; //Reset for the next time
-            } else {
-              return result.then(() => {
+          let func = this.view(0) ? this.view(0).firstChild.au['virtual-repeat-next'].instruction.attributes['virtual-repeat-next'] : undefined;
+          let topIndex = this._first;
+          let isAtBottom = this._bottomBufferHeight === 0;
+          let isAtTop = this._isAtTop;
+
+          this.scope.overrideContext.$topIndex = topIndex;
+          this.scope.overrideContext.$isAtBottom = isAtBottom;
+          this.scope.overrideContext.$isAtTop = isAtTop;
+
+          if (func === undefined) {
+            return null;
+          } else if (typeof func === 'string') {
+            let getMoreFuncName = this.view(0).firstChild.getAttribute('virtual-repeat-next');
+            let funcCall = this.scope.overrideContext.bindingContext[getMoreFuncName];
+
+            if (typeof funcCall === 'function') {
+              let result = funcCall.call(this.scope.overrideContext.bindingContext, topIndex, isAtBottom, isAtTop);
+              if (!(result instanceof Promise)) {
                 this._calledGetMore = false; //Reset for the next time
-              });
+              } else {
+                return result.then(() => {
+                  this._calledGetMore = false; //Reset for the next time
+                });
+              }
+            } else {
+              throw new Error("'virtual-repeat-next' must be a function or evaluate to one");
             }
+          } else if (func.sourceExpression) {
+            return func.sourceExpression.evaluate(this.scope);
```

Comment:
Yes, thank you for catching this. I missed the case of flagging a completed "getMore" event when using call.